terraform/modules/security/main.tf

```diff
@@ -1,41 +1,23 @@
 resource "azurerm_network_security_group" "nsg" {
-  name                = var.nsg_name
+  count               = length(var.security_groups)
```

Comment:
Daca ar exista un map separat de security group (in loc sa fie totul la gramada) aici ai putea sa faci for_each si sa iterezi prin map mult mai simplu decat cu count